src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -16,383 +15,264 @@ public static class NUnitTestExtractorApp
-                                    }
-
-                                    string data = string.Empty;
-
-                                    //If exclude is true then move on to next method as this one has category marked as exluded
-                                    if (exclude)
-                                    {
-                                        continue;
-                                    }
-                                    else if (include)
-                                    {
-                                        data = FormattedTest(level, type, methodInfo, testCase, data);
-
-                                        //Prevent duplicate entries from being written
-                                        if (!string.IsNullOrEmpty(data) && !testsWritten.Contains(data))
-                                        {
-                                            WriteTestDllAndName(writer, dll, data);
-                                            testsWritten.Add(data);
-                                        }
-                                    }
-                                }
-                            }
-                            //reset categoryNames for next method
-                            categoryNames = new List<string>();
-                        }
-                    }
-                }
-            }
-            else
+            // if the parent directory doesn't exist, then create it
+            if (!fi.Directory.Exists)
             {
-                throw new ArgumentNullException("dlls", "no dlls were specified");
+                fi.Directory.Create();
             }
         }
 
         /// <summary>
-        /// Returns the properly formatted test string depending on the level specified. 
+        /// Get a list of namespaces/classes/functions/testcases from the assembly
         /// </summary>
-        /// <param name="level">Used in the switch statement to determine how much info to put into the data string</param>
-        /// <param name="type">Used to get the namespace of the test</param>
-        /// <param name="methodInfo">Used to get the test info such as declaring type and name</param>
-        /// <param name="testCase">Used to get arguments of a testcase</param>
-        /// <param name="data">The string which is modified and then returned</param>
-        /// <returns>Returns properly formatted test string</returns>
-        private static string FormattedTest(Level level, Type type, MethodInfo methodInfo, TestCaseAttribute testCase, string data)
+        /// <param name="assemblyFullPath"></param>
+        /// <returns></returns>
+        public static void GenerateInfoFromAssembly(string assemblyFullPath)
         {
-            switch (level)
-            {
-                case Level.Namespace:
-                    data = type.Namespace;
-                    break;
+            HashSet<string> rv = new HashSet<string>();
```

Comment:
[Should] Use a more descriptive name than rv...

Reply:
Fixed